src/placeos-rest-api/controllers/systems.cr

```diff
@@ -202,6 +203,14 @@ module PlaceOS::Api
       render json: documents
     end
 
+    # Return metadata for the system
+    #
+    get "/:sys_id/metadata", :metadata do
+      parent_id = current_control_system.id.not_nil!
+      name = params["name"]?.presence
+      render json: Model::Metadata.build_metadata(parent_id, name)
+    end
+
```

Comment:
Could this be extracted into a module? I guess the `id_param` is an issue.

Reply:
Yep - that was the reason for not. It's a solvable issue, but was trying to avoid a wider refactor. Also arguably a little clearer having these sitting directly in the controllers.